src/Cosmos.Coroutines/System/Coroutines/Coroutine.cs

```diff
@@ -42,13 +42,25 @@ internal void Step()
         /// Starts the execution of this coroutine on the global coroutine pool.
         /// </summary>
         /// <exception cref="InvalidOperationException">Thrown when an attempt is made to start the coroutine while it's already running.</exception>
-        public void Start()
+        /// <param name="autoJoin">if false will not auto join main pool.</param>
+        public void Start(bool autoJoin = true)
         {
             if(Running) {
                 throw new InvalidOperationException("Cannot start an already running Coroutine.");
             }
 
-            CoroutinePool.Main.AddCoroutine(this);
+            if (autoJoin && Pool == null)
+            {
+                CoroutinePool.Main.AddCoroutine(this);
+            }
+
+            if (Pool == null)
+            {
+                throw new InvalidOperationException("Cannot start Coroutine not in a pool.");
```

Comment:
I think this might confuse some devs. I feel like "Cannot start the execution of the Coroutine on a set pool, with the Pool property being set to null." would be better